development/tools/pkg/synchronizer/syncomponent/component_test.go

```diff
@@ -0,0 +1,147 @@
+package syncomponent
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSynComponent(t *testing.T) {
+	// Given
+	path := "/this/is/main/path/this/is/comp-name"
+	mainPath := "/this/is/main/path"
+	componentPath, _ := filepath.Rel(mainPath, path)
+
+	// When
+	component := NewSynComponent(componentPath, []string{"/path/to/resource"})
+
+	// Then
+	assert.Equal(t, "comp_name", component.Name)
+	assert.Equal(t, "this/is/comp-name", component.Path)
+	assert.Equal(t, "/path/to/resource", component.Versions[0].VersionPath)
+}
+
+func TestComponent_AddGitHashHistory(t *testing.T) {
+	// Given
+	key1 := subtractDaysFromToday(3)
+	key2 := subtractDaysFromToday(2)
+	key3 := subtractDaysFromToday(0)
+	key4 := subtractDaysFromToday(1)
+
+	history := map[int64]string{
+		key1: "a1a1a1a1",
+		key3: "c3c3c3c3",
+		key2: "b2b2b2b2",
+		key4: "d4d4d4d4",
+	}
+
+	component := NewSynComponent(".", []string{"."})
+
+	// When
+	component.GitHashHistory = history
+
+	// Then
+	assert.Equal(t, "a1a1a1a1", component.GetOldest())
+}
+
+func Test_isOutOfDate(t *testing.T) {
+	// Given
+	tsTrue := []Component{
+		{
+			GitHash: "c33e5390",
+			GitHashHistory: map[int64]string{
+				subtractDaysFromToday(5): "4e1e6950",
+			},
+			Versions: []*ComponentVersions{
+				{
+					Version:       "b28250b7",
+					ModifiedFiles: []string{"/path/to/file.go"},
+				},
+			},
+		},
+		{
+			GitHash: "06e635f8",
+			GitHashHistory: map[int64]string{
+				subtractDaysFromToday(2): "06e635f8",
+				subtractDaysFromToday(6): "a00c9f2d",
+			},
+			Versions: []*ComponentVersions{
+				{
+					Version:       "06e635f8",
+					ModifiedFiles: []string{},
+				},
+				{
+					Version:       "b28250b7",
+					ModifiedFiles: []string{"/path/to/file"},
+				},
+			},
+		},
+	}
+
+	tsFalse := []Component{
+		{
+			// is not out of date because component version contained in history logs
+			GitHash: "c33e5390",
+			GitHashHistory: map[int64]string{
+				subtractDaysFromToday(0): "86d0ddac3df8e7c4ef77f09158c68fa06566537d",
+				subtractDaysFromToday(1): "b28250b72854bae58398fa599990a70d3844ec8f",
+				subtractDaysFromToday(2): "c33e539019a9fca5332501098527ba05c0084973",
+			},
+			Versions: []*ComponentVersions{
+				{
+					Version:       "b28250b7",
+					ModifiedFiles: []string{"/path/to/file.go"},
+				},
+			},
+		},
+		{
+			// is not out of date because currrent hash is equal to version
+			GitHash: "a30ed000",
+			GitHashHistory: map[int64]string{
+				subtractDaysFromToday(1): "fc74b6ed1764c178db8ba0f78736dd9830625b37",
+			},
+			Versions: []*ComponentVersions{
+				{
+					Version:       "a30ed000",
+					ModifiedFiles: []string{},
+				},
+			},
+		},
+		{
+			//is not out of date because only md file was changed
```

Comment:
I think it will be more readable if you get rid of the slices tsTrue and tsFalse and write:
```
t.Run("is not out of date because only md file was changed", func(t *testing.T) ... 
```

Reply:
:+1: